river/Cursor.zig

```diff
@@ -162,9 +184,16 @@ pub fn init(self: *Self, seat: *Seat) !void {
     wlr_cursor.events.pinch_update.add(&self.pinch_update);
     wlr_cursor.events.pinch_end.add(&self.pinch_end);
     seat.wlr_seat.events.request_set_cursor.add(&self.request_set_cursor);
+    // Ensure that it is safe to call remove()
+    self.saved_image_surface_destroy.link = .{
+        .prev = &self.saved_image_surface_destroy.link,
+        .next = &self.saved_image_surface_destroy.link,
+    };
 }
 
 pub fn deinit(self: *Self) void {
+    self.auto_hide_timer.remove();
+    // why are no listeners removed?
```

Comment:
Is there some logic behind this, or are no listeners removed because it hasn’t triggered any segfault yet? Should i remove `saved_image_surface_destroy`?